mlflow/models/evaluation/base.py

```diff
@@ -1102,6 +1106,8 @@ def evaluate(
                    are regarded as feature columns. If it is Spark DataFrame, only the first 10000
                    rows in the Spark DataFrame will be used as evaluation data.
 
+                - A MLflow Dataset object, containing evaluation features and labels.
```

Comment:
`targets` should be optional if an MLflow Dataset object is specified.

Reply:
If an MLflow Dataset is specified but doesn't define `targets`, we should raise a clear error message indicating that either the dataset needs to specify targets or the user needs to pass a value for the targets param of `mlflow.evaluate()`.